src/render/software/SDL_render_sw.c

```diff
@@ -500,10 +491,14 @@ SW_RenderCopyEx(SDL_Renderer * renderer, SDL_Surface *surface, SDL_Texture * tex
                          * to be created. This makes all source pixels opaque and the colors get copied correctly.
                          */
                         SDL_Surface *src_rotated_rgb;
-                        src_rotated_rgb = SDL_CreateRGBSurfaceFrom(src_rotated->pixels, src_rotated->w, src_rotated->h,
-                                                                   src_rotated->format->BitsPerPixel, src_rotated->pitch,
-                                                                   src_rotated->format->Rmask, src_rotated->format->Gmask,
-                                                                   src_rotated->format->Bmask, 0);
+                        int f = SDL_MasksToPixelFormatEnum(src_rotated->format->BitsPerPixel,
+                                            src_rotated->format->Rmask,
+                                            src_rotated->format->Gmask,
+                                            src_rotated->format->Bmask,
+                                            0);
+
+                        src_rotated_rgb = SDL_CreateRGBSurfaceWithFormatFrom(src_rotated->pixels, src_rotated->w, src_rotated->h, 
+                                                                             src_rotated->pitch, f);
```

Comment:
Does SDLgfx_rotateSurface() not set src_rotated->format->format?

Reply:
it does, but here it seems we want src_rotated format with AMask forced to 0

Reply:
Gotcha